### PR TITLE
chore: updated typesript version to 4.1.x

### PR DESCRIPTION
**Pull request by karandashovrz**
affects: @cuba-platform/react-core, @cuba-platform/react-ui, @cuba-platform/rest,
@cuba-platform/front-generator

### CODE REVIEW

**Comment by vyacheslav-pushkin** [comment]
@karandashovrz CI tests fail, probably there is a breaking change in the new TS that results in compile error in front-generator:

```
src/generators/sdk/model/entities-generation.ts(191,30): error TS2345: Argument of type 'SyntaxKind.NullKeyword' is not assignable to parameter of type 'KeywordTypeSyntaxKind'.
```

Please look into this. Thanks.

File: packages/cuba-react-core/package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "tslint-config-prettier": "^1.18.0",
     "tslint-react": "^4.0.0",
     "typedoc": "^0.15.3",
-    "typescript": "~3.7.4"
+    "typescript": "~4.1.x"
```
Comment (vyacheslav-pushkin) [inline packages/cuba-react-core/package.json:43]: Nitpick: I'd use either `~` or `.x` but not both as they have the same meaning in this case.

**Review by vadimbasko — APPROVED** [review]

File: packages/cuba-react-core/package.json (hunk 1 above)
Comment (karandashovrz) [reply packages/cuba-react-core/package.json]: > @karandashovrz CI tests fail, probably there is a breaking change in the new TS that results in compile error in front-generator:
> 
> ```
> src/generators/sdk/model/entities-generation.ts(191,30): error TS2345: Argument of type 'SyntaxKind.NullKeyword' is not assignable to parameter of type 'KeywordTypeSyntaxKind'.
> ```
> 
> Please look into this. Thanks.

@vyacheslav-pushkin fixed

Comment (karandashovrz) [reply packages/cuba-react-core/package.json]: > Nitpick: I'd use either `~` or `.x` but not both as they have the same meaning in this case.

@vyacheslav-pushkin fixed

**Review by vyacheslav-pushkin — APPROVED** [review]

**Pull request merged**